### ISSUE TITLE
admin.groups missing 1 required positional argument: 'top'

### ISSUE THREAD

**Issue by caroljenner** [andrewvillazon/pbipy#9]
When we run the admin.groups()   we get this error:

```
TypeError    Traceback (most recent call last) Cell In [57], line 1----> 1 zEvents=admin.groups()TypeError: Admin.groups() missing 1 required positional argument: 'top'
```

Is this an issue or is groups handled differently?

**Comment by andrewvillazon** [comment]
Looking in a bit more detail, the [Admin GetGroups](https://learn.microsoft.com/en-us/rest/api/power-bi/admin/groups-get-groups-as-admin) endpoint requires a `top` parameter so this was reflected in the `admin.groups()` method. But thinking about it, making it optional (like the non-admin [GetGroups](https://learn.microsoft.com/en-us/rest/api/power-bi/groups/get-groups) endpoint) would improve usability. I'll change the method and push an update in the next day or two.

In the meantime, try including a value for the `top` argument and it should work. The api docs suggest 5000 is the limit on this.

**Comment by andrewvillazon** [comment]
@caroljenner - I've made the `top` parameter optional, so calling `admin.groups()` without `top` should no longer get the error. Have made available on this branch if you want to have a look.

```
pip install git+https://github.com/andrewvillazon/pbipy@feature/9/admin-groups-missing-1-required-positional-argument-top
```

**Comment by caroljenner** [comment]
Thanks!  I’ll give it to the developer and get back to you!

Thank you again on the quick response.


Carol Jenner
Data Architect

***@***.******@***.***> - patternenergy.com<https://patternenergy.com>


Read how Pattern is transitioning the world to renewable energy in our latest Sustainability Report<https://bit.ly/PatternForSustainability>

             [https://patternenergy.com/wp-content/uploads/2022/04/Sustainabitlity-leaf-icon-3.png]



...
This email message may contain information that is confidential and proprietary. If you are not the intended recipient, please contact the sender and destroy the original and any copies of the original message. We take measures to protect the content of our communications. However, we cannot guarantee that email messages will not be intercepted by third parties or that email messages will be free of errors or viruses.

**Comment by andrewvillazon** [comment]
Changes merged and released under [v2.6.0](https://github.com/andrewvillazon/pbipy/releases/tag/v2.6.0). Closing.
